Week1 [STEP 3] 비모

bubblecocoa opened the pull request:
@JoSH0318 
안녕하세요 밈👋
Step 3: 주문에 따른 빼빼로 그리기 제출합니다!

## 고민했던 점
### 빼빼로는 어떻게 생겼을까? 🔍
사전적 정의는 아니지만 개인적인 생각으로 빼빼로는 막대 과자의 외부나 내부에 초코 소스를 기반으로 토핑을 추가할 수 있는 과자라고 생각합니다. 막대 형태의 과자가 기반이 되기 때문에 몸통이라고 볼 수 있는 부분이 막대라고 볼 수 있는 부분보다 얇아지는 일은 없는 경우가 일반적일 것입니다.
### 이건 빼빼로가 맞을까? 🤔
하지만 예시로 주어진 1~4번 케이스의 경우 모두 몸통이 막대보다 얇은 케이스는 없었습니다. if문을 통해 몸통이 빼빼로보다 얇은 경우 "빼빼로가 아닙니다."라는 문구를 출력해볼까? 했으나 요구사항 어디에도 빼빼로의 형태를 검증하라는 말은 없어 생략하였습니다.
### 몸통과 막대의 두께가 같은 빼빼로 🍭
예시 1번(일반 빼빼로)과 예시 4번(누드 뺴뺴로)의 경우 빼빼로의 좌측에 공백이 1개 추가되었다는 설명이 있습니다. 일괄적으로 공백이 추가가 된것도 아니고 일반 빼빼로와 누드 빼빼로의 경우에만 공백이 추가되었으니 해당 케이스가 어떤 경우인지 자세히 확인해 보았습니다. 여러가지 관점으로 볼 수 있겠으나, 저는 몸통부분과 막대 부분의 두께가 같은 경우 좌측의 공백이 1개 추가된다고 판단했습니다.
### 일반적으로 비대칭인 빼빼로는 없다 😵
만들기 나름이겠으나, 일반적으로 빼빼로는 정면으로 바라보았을때 좌우가 거의 대칭일 것입니다. 예시 3, 4를 보면 막대 부분이 몸통의 중앙 부분에 위치하고 있습니다. 일반 빼빼로와 누드 빼빼로가 아닌 경우에는 막대의 공백만 계산해야 된다는 의미로 해석했습니다.
### 막대 공백 계산 🧮
빼빼로의 몸통과 토핑이 얼마나 두꺼워지느냐에 따라 얼마나 공백을 주어야 할지 계산해야 합니다. 저는 `(몸통+토핑의 두께 - 막대의 두께) / 2`를 막대가 그려지기 전 그려져야 할 공백으로 계산했습니다. 여러번의 실행 결과 두께가 짝수인 경우는 정확히 가운데 위치할 수 없지만 몸통+토핑의 두께와 막대의 두께가 바뀌어도 막대가 정상적으로 가운데에 위치하는것을 확인했습니다.

## 조언을 얻고 싶은 부분
### nil
일반 빼빼로나 누드 빼빼로의 경우 토핑이 없는데, 저는 매개변수를 `""`로 전달했습니다. 해당 문제에서 `nil`로 전달될수도 있는걸까요? `nil`을 다루는 법은 `옵셔널`파트에서 다룬다고 했는데 사용해본적이 없는 개념이라 어렵네요 🥵

bubblecocoa committed:
Step 3: 주문에 따른 빼빼로 그리기

JoSH0318 reviewed (commented):
@bubblecocoa 
코드를 통해 이번 step에서 많은 고민을 하신 게 보입니다 😀
이런 저런 고민을 하신 모습이 너무 멋지다고 생각해요 👍

pr에 대한 코멘트 남겼습니다 😀

---
> Q: 일반 빼빼로나 누드 빼빼로의 경우 토핑이 없는데, 저는 매개변수를 ""로 전달했습니다. 해당 문제에서 nil로 전달될수도 있는걸까요? nil을 다루는 법은 옵셔널파트에서 다룬다고 했는데 사용해본적이 없는 개념이라 어렵네요 🥵

A: nil에 대해서 알아보셨군요 👍 nil이 swift 안정성에 크게 기여하는 특별한 점이기도 합니다. 
하지만, 아직 nil을 학습하기엔 조금 이를 수도 있다는 생각이 들어서 조언을 하기가 어려운 부분이 있어요 🥹
그래도 nil에 대해 학습한 후에 이야기를 나누는 것이 좋을 것 같아요 ! 
이부분에 대해서 궁금하시다면 DM 주시면 최대한 답변하겠습니다 😀

bubblecocoa committed:
fix: Step 3: 주문에 따른 빼빼로 그리기

변수명, 함수명 약자 사용 X
가시성 향상을 위해 calcSpaceCount 함수를 makeToppingBodySpace, makeStickSpace 두 함수로 분리

JoSH0318 reviewed (commented):
@bubblecocoa

요구에 따라 수정해주시고 고민해주셔서 감사합니다 😀
commit 제목에 fix를 붙여주신 것을 보니 commit convention을 만드신 것 같아 보이네요 👍
이점 너무 멋집니다 👍

특별히 더 수정할 부분은 없고, 추가 코멘트한 것은 사소한 부분이니 
일단 merge 하도록 하겠습니다!

---

> Q: 밈이 활용하시는 검색 방법이 있을까요? 

A: 지극히 개인적인 견해입니다 😀 
저는 `swift string multiline docs` 이런식으로 검색해서 보는 편입니다.
서칭 능력도 실력인 것을 느낄때가 많습니다. 🥹
아직 저도 좀 더 나은 방법을 찾고 있는 중입니다.(사실 평생 그럴것 같아요🥹)
